Job-DSL Issue

Job DSL does not have `Extension-Name` in `MANIFEST.MF` which might be the cause of:

```
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Update center generator for jenkins-ci.org 1.21-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] >>> exec-maven-plugin:1.2:java (default-cli) @ update-center2 >>>
[INFO] 
[INFO] --- maven-enforcer-plugin:1.0.1:display-info (default) @ update-center2 ---
[INFO] Maven Version: 3.0.5
[INFO] JDK Version: 1.7.0_75 normalized as: 1.7.0-75
[INFO] OS Info: Arch: amd64 Family: unix Name: linux Version: 3.18.5-x86_64-linode52
[INFO] 
[INFO] <<< exec-maven-plugin:1.2:java (default-cli) @ update-center2 <<<
[INFO] 
[INFO] --- exec-maven-plugin:1.2:java (default-cli) @ update-center2 ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.330s
[INFO] Finished at: Wed Mar 18 19:49:46 UTC 2015
[INFO] Final Memory: 13M/93M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2:java (default-cli) on project update-center2: An exception occured while executing the Java class. null: InvocationTarge
tException: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2:java (default-cli) on project update-center2: An exception occured while
 executing the Java class. null
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:217)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:320)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:414)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:357)
Caused by: org.apache.maven.plugin.MojoExecutionException: An exception occured while executing the Java class. null
        at org.codehaus.mojo.exec.ExecJavaMojo.execute(ExecJavaMojo.java:346)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
        ... 19 more
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.codehaus.mojo.exec.ExecJavaMojo$1.run(ExecJavaMojo.java:291)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
        at java.lang.String$CaseInsensitiveComparator.compare(String.java:1176)
        at java.lang.String$CaseInsensitiveComparator.compare(String.java:1170)
        at java.util.TreeMap.getEntryUsingComparator(TreeMap.java:369)
        at java.util.TreeMap.getEntry(TreeMap.java:340)
        at java.util.TreeMap.get(TreeMap.java:273)
        at org.jvnet.hudson.update_center.LocalDirectoryRepository.listHudsonPlugins(LocalDirectoryRepository.java:132)
        at org.jvnet.hudson.update_center.Main.buildPlugins(Main.java:274)
        at org.jvnet.hudson.update_center.Main.buildUpdateCenterJson(Main.java:203)
        at org.jvnet.hudson.update_center.Main.run(Main.java:167)
        at org.jvnet.hudson.update_center.Main.run(Main.java:144)
        at org.jvnet.hudson.update_center.Main.main(Main.java:131)
        ... 6 more
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

On missing `Extension-Name` I tried setting that to `Short-Name`.
On missing `Specification-Title` I set it to an empty string.

However, the execution stops in the middle
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.983s
[INFO] Finished at: Wed Mar 18 20:06:55 UTC 2015
[INFO] Final Memory: 16M/93M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2:java (default-cli) on project update-center2: An exception occured while executing the Java class. null: InvocationTarge
tException: entry -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2:java (default-cli) on project update-center2: An exception occured while
 executing the Java class. null
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:217)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:320)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:414)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:357)
Caused by: org.apache.maven.plugin.MojoExecutionException: An exception occured while executing the Java class. null
        at org.codehaus.mojo.exec.ExecJavaMojo.execute(ExecJavaMojo.java:346)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
        ... 19 moreCaused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.codehaus.mojo.exec.ExecJavaMojo$1.run(ExecJavaMojo.java:291)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException: entry
        at java.util.zip.ZipFile.getInputStream(ZipFile.java:342)
        at java.util.jar.JarFile.getInputStream(JarFile.java:412)
        at org.jvnet.hudson.update_center.LocalHPI.resolvePOM(LocalHPI.java:83)
        at org.jvnet.hudson.update_center.Plugin.readPOM(Plugin.java:157)
        at org.jvnet.hudson.update_center.Plugin.<init>(Plugin.java:127)
        at org.jvnet.hudson.update_center.Main.buildPlugins(Main.java:278)
        at org.jvnet.hudson.update_center.Main.buildUpdateCenterJson(Main.java:203)
        at org.jvnet.hudson.update_center.Main.run(Main.java:167)
        at org.jvnet.hudson.update_center.Main.run(Main.java:144)
        at org.jvnet.hudson.update_center.Main.main(Main.java:131)
        ... 6 more
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/src/main/java/org/jvnet/hudson/update_center/LocalDirectoryRepository.java
+++ b/src/main/java/org/jvnet/hudson/update_center/LocalDirectoryRepository.java
@@ -106,12 +106,14 @@ public class LocalDirectoryRepository extends MavenRepository
             
             final String extension = FilenameUtils.getExtension(filename);
 
+            // Extension-Name and Implementation-Title is not set for plugin using gradle (ex: ivy:1.26)
+            // Short-Name seems always present and have the same value as the other two.
+            String artifactId = manifest.getMainAttributes().getValue("Short-Name");
             ArtifactInfo a = new ArtifactInfo(
                     null,  // fname
                     extension,
                     groupId,
-                    manifest.getMainAttributes().getValue("Extension-Name"),    // artifactId
-                        // maybe Short-Name or Implementation-Title is more proper.
+                    artifactId,
                     manifest.getMainAttributes().getValue("Plugin-Version"),    // version
                         // maybe Implementation-Version is more proper.
                     null,  // classifier

